src/providers/ldap/sdap_async_users.c

```diff
@@ -269,7 +272,24 @@ int sdap_save_user(TALLOC_CTX *memctx,
     DEBUG(SSSDBG_TRACE_FUNC, "Processing user %s\n", user_name);
 
     if (opts->schema_type == SDAP_SCHEMA_AD) {
+	/* construct canonical UPN from sAMAccountName to help Samba and also to allow us to lookup user via UPN */
         ret = sysdb_attrs_get_string(attrs,
+                    opts->user_map[SDAP_AT_USER_SAMACCOUNTNAME].sys_name, &samaccountname);
+	if (ret == EOK) {
+	    ret = ENOENT;
+	    realm = dp_opt_get_cstring(opts->basic, SDAP_KRB5_REALM);
```

Comment:
Hi,

I think it would be better to use `realm = dom->realm` here, `opts->basic` contain the global options of the backend and `SDAP_KRB5_REALM` is the realm of the domain you joined to. If the user is coming from a trusted domain this realm would be wrong.

bye,
Sumit